common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java

```diff
@@ -284,7 +284,7 @@ public UUID createSchema(String schemaName,
                              Map<String, String> metadata) throws AWSSchemaRegistryException {
         UUID schemaVersionId = null;
         try {
-            log.info("Auto Creating schema with schemaName: {} and schemaDefinition : {}", schemaName,
```

Comment:
Are you seeing this logged for every request?

Reply:
We're only seeing the "Auto Creating schema" message once, and then the "Schema Version Id is null" on every message.

Reply:
Thanks for fixing this. Would you mind updating it in class [GlueSchemaRegistryKafkaSerializer.java](https://github.com/awslabs/aws-glue-schema-registry/blob/master/serializer-deserializer/src/main/java/com/amazonaws/services/schemaregistry/serializers/GlueSchemaRegistryKafkaSerializer.java) as well. This class is a generic class that can be used with additional property called "dataFormat" and passing "JSON" or "AVRO".

Reply:
Note that re the information in #63, I think there might be an underlying issue, and changing the log level will just mask that.

Reply:
I think this got lost in merge conflicts resolution. This has been a miss on our side. The schema auto creation should still be info because we don't expect that to happen very often. The Over logging issue is legit. Are you suggesting that there is an underlying issue with the library or your set-up particularly ? Either ways, we would be happy to help.

Reply:
Well, we're getting the "_Schema Version Id is null. Trying to register the schema_" log for every serialized message (which is fun when doing initial dump of a 200+ million row table 😀) – I'm *assuming* the schema identifier ought to be saved after (auto-)creation, but I don't know if it's an issue with the serializer, our configuration, or the interplay between Debezium and Kafka Connect.

As mentioned in #63, the caching in `GlueSchemaRegistrySerializationFacade` seems to be working, so we don't end up swamping Glue SR with requests – but my "_I haven't studied the code enough to understand it well_" hunch is that it probably shouldn't be the cache layer preventing this?

Reply:
That logging message should be debug. Some customers have just one schemaVersionId and that can be hard coded in the AWSKafkaAvroSerializer for efficiency. Schema Identifier for your case would be saved in a in-memory cache later in the code path and the message you are seeing may confuse you that schema version id is null.

Reply:
I _think_ @snemarch means, that there is no way to specify the `schemaVersionId` when using Kafka Connect - or documentation around when that would be applicable. 🙂

Reply:
I've updated `GlueSchemaRegistryKafkaSerializer` to also use `debug` for the logging is `serialize`.

Reply:
@mohitpali - so, if I understand correctly:

1. Usecases where dynamic schema registration is used isn't meant to store the registered schemaVersionId, and the cache a crucial part of the design, not "just" an optimization.
2. Usecases where you can manually evolve schemas can have the schemaVersionId predefined.

Even with re-changing log level to DEBUG, would it be worth adding a comment to the codebase that it's normal to get this on a per-message basis for dynamic scenarios, and that caching further down ensures this is not a performance problem?

Thanks for your help in clearing up matters! :)

Reply:
Good point, we can add these comments, thanks !

Reply:
If you could please change the below line to info, we should be good to merge.
``
log.debug("Auto Creating schema with schemaName: {} and schemaDefinition : {}", schemaName, schemaDefinition);
``

Reply:
> If you could please change the below line to info, we should be good to merge.

@tveon is on vacation for the next few weeks, but I'll see if I can persuade him to do that one-line update :)

Reply:
Done. 👍 🌴